homeassistant/components/ambient_network/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""DataUpdateCoordinator for the Ambient Weather Network integration."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any, cast
+
+from aioambient import OpenAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import API_LAST_DATA, DOMAIN, ENTITY_MAC_ADDRESS, LOGGER, SCAN_INTERVAL
+
+
+class AmbientNetworkDataUpdateCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """The Ambient Network Data Update Coordinator."""
+
+    config_entry: ConfigEntry
+
+    def __init__(self, hass: HomeAssistant, api: OpenAPI) -> None:
+        """Initialize the coordinator."""
+        super().__init__(hass, LOGGER, name=DOMAIN, update_interval=SCAN_INTERVAL)
+        self.api = api
+        self.data = {}
+
+    async def _async_update_data(self) -> dict[str, Any]:
```

Comment:
Would it make sense to raise UpdateFailed whenever its the first update? This might be a more user friendly way of saying something is wrong

Reply:
I will keep that for a follow-up PR. We can generally make this more robust by logging failures on first error and logging when the station comes back. For now, I want to keep things simple to land a first version.

Reply:
Ok. I have added error handling and logging.